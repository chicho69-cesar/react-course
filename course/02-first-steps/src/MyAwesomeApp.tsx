import type { CSSProperties } from 'react'

const firstName = 'Cesar'
const lastName = 'Villalobos Olmos'

const favoriteGames = ['Fifa', 'Call of Duty', 'Grand Theft Auto V', 'God of War Ragnarok']
const isActive = false

const address = {
  zinCode: 'ABC-123',
  country: 'Mexico',
}

const styles: CSSProperties = {
  backgroundColor: '#fafafa',
  borderRadius: 20,
  padding: 10,
  marginTop: 30,
}

export default function MyAwesomeApp() {
  return (
    <div data-testid='div-app'>
      <h1 data-testid='first-name-title'>{firstName}</h1>
      <h3>{lastName}</h3>

      <p>{favoriteGames.join(', ')}</p>
      <h2>{isActive ? 'Activo' : 'No Activo'}</h2>

      <p style={styles}>
        {JSON.stringify(address, null, 2)}
      </p>
    </div>
  )
}
