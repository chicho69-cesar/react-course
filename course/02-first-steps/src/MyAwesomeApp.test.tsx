import { render, screen } from '@testing-library/react'
import { describe, expect, test } from 'vitest'
import MyAwesomeApp from './MyAwesomeApp'

describe('MyAwesomeApp', () => {
  test('Should render the firstName and the lastName', () => {
    const { container } = render(<MyAwesomeApp />)

    const h1 = container.querySelector('h1')
    const h3 = container.querySelector('h3')

    expect(h1?.innerHTML).toContain('Cesar')
    expect(h3?.innerHTML).toContain('Villalobos Olmos')
  })

  test('Should render the firstName and the lastName with screen', () => {
    render(<MyAwesomeApp />)
    screen.debug()

    const name = screen.getByTestId('first-name-title')
    expect(name.innerHTML).toContain('Cesar')
  })

  test('Should match snapshot', () => {
    const { container } = render(<MyAwesomeApp />)
    expect(container).toMatchSnapshot()
  })

  test('Should match snapshot', () => {
    render(<MyAwesomeApp />)
    expect(screen.getByTestId('div-app')).toMatchSnapshot()
  })
})
