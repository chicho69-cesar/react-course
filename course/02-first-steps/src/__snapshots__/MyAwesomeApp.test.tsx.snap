// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Should match snapshot 1`] = `
<div>
  <div
    data-testid="div-app"
  >
    <h1
      data-testid="first-name-title"
    >
      Cesar
    </h1>
    <h3>
      Villalobos Olmos
    </h3>
    <p>
      Fifa, Call of Duty, Grand Theft Auto V, God of War Ragnarok
    </p>
    <h2>
      No Activo
    </h2>
    <p
      style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
    >
      {
  "zinCode": "ABC-123",
  "country": "Mexico"
}
    </p>
  </div>
</div>
`;

exports[`MyAwesomeApp > Should match snapshot 2`] = `
<div
  data-testid="div-app"
>
  <h1
    data-testid="first-name-title"
  >
    Cesar
  </h1>
  <h3>
    Villalobos Olmos
  </h3>
  <p>
    Fifa, Call of Duty, Grand Theft Auto V, God of War Ragnarok
  </p>
  <h2>
    No Activo
  </h2>
  <p
    style="background-color: rgb(250, 250, 250); border-radius: 20px; padding: 10px; margin-top: 30px;"
  >
    {
  "zinCode": "ABC-123",
  "country": "Mexico"
}
  </p>
</div>
`;
